fdb-relational-core/src/test/java/com/apple/foundationdb/relational/recordlayer/RelationalArrayTest.java

```diff
@@ -124,18 +125,14 @@ private void insertArraysViaQuerySimpleStatement() throws SQLException {
                         "[x'31', x'32'], null, " +
                         "[(11, '11'), (22, '22')], null " +
                         ")"));
-        // This should not go through since non-null array is not initialized, However, that is being currently getting
-        // initialized by an empty array (no elements in unwrapped REPEATED field in proto).
-        // This phenomenon is also true with basic types, i.e., non-nullability is not enforced and is tracked by
-        // TODO (Add support + tests for column nullable/not null)
-        insertQuery("INSERT INTO T (pk) VALUES (4)");
+        Assert.assertThrows(SQLException.class, () -> insertQuery("INSERT INTO T (pk) VALUES (4)"));
     }
 
     @Test
     void testInsertArraysViaQueryPreparedStatement() throws SQLException {
         final var statement = "INSERT INTO T (pk, boolean_null, boolean_not_null, integer_null, integer_not_null, " +
                 "bigint_null, bigint_not_null, float_null, float_not_null, double_null, double_not_null, string_null, " +
-                "string_not_null, bytes_null, bytes_not_null, struct_null, struct_not_null) VALUES (?pk, ?boolean_null, " +
+                "string_not_null, bytes_null, bytes_not_null, structure_null, structure_not_null) VALUES (?pk, ?boolean_null, " +
```

Comment:
You probably want to revert back to `struct_null` instead of `structure_null` since the type name is `STRUCT`

Reply:
done!